### PR TITLE
Classes to work with user data provided by http headers

### PR DESCRIPTION
**Pull request by myluki2000**
## Description of changes
This adds some classes used to read user data from the http headers that are set by the gateway.

The `LoggedInUser` class is the data class which contains the user information extracted from the http headers.

The `RequestHeaderUserProcessor` class should be used in a `@Component WebGraphQlInterceptor` implementation in a service to extract and convert the http headers from http requests sent to the service and inject a `LoggedInUser` object into the graphql context of the service.

  
## How has this been tested?
Please describe the test strategy you followed.

- [x] automated unit test
- [ ] automated integration test
- [ ] automated acceptance test
- [x] manual, exploratory test

Correct deserialization is tested by sending a request with the CurrentUser header set to a JSON serialized version of the LoggedInUser. RequestHeaderUserProcessor should then deserialize the LoggedInUser data correctly.
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [x] I have made corresponding changes to the documentation
 [YES!](https://gits-enpro.readthedocs.io/en/latest/dev-manuals/backend/auth%26accessing-user-data.html)
- [ ] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [ ] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/java/de/unistuttgart/iste/gits/common/user_handling/RequestHeaderUserProcessor.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ */
+public class RequestHeaderUserProcessor {
+    /**
+     * Call this method from your WebGraphQlInterceptor implementation to extract the current user from the request and
+     * add the user data to the graphql context as a LoggedInUser object living under the key "currentUser".
+     * @param request The request to process.
+     */
+    @SneakyThrows
+    public static void process(WebGraphQlRequest request) {
+        if(!request.getHeaders().containsKey("CurrentUser"))
```
Comment (ZombieAlienRobot) [inline src/main/java/de/unistuttgart/iste/gits/common/user_handling/RequestHeaderUserProcessor.java:31]: Should add {} to the if. Makes it easier to understand and prevents unwanted side effects.

**Review by ZombieAlienRobot — CHANGES_REQUESTED** [review]

File: src/main/java/de/unistuttgart/iste/gits/common/user_handling/RequestHeaderUserProcessor.java (hunk 1 above)
Comment (myluki2000) [reply src/main/java/de/unistuttgart/iste/gits/common/user_handling/RequestHeaderUserProcessor.java]: I disagree but I changed it

**Review by ZombieAlienRobot — APPROVED** [review]
